custom/enikshay/integrations/nikshay/repeater_generator.py

```diff
@@ -417,36 +417,35 @@ def get_payload(self, repeat_record, episode_case):
         }
 
     def handle_success(self, response, payload_doc, repeat_record):
-        # A successful response returns a Nikshay ID like 00001
-        # Failures also return with status code 200 and some message like
-        # Dublicate Entry or Invalid data format
-        # (Dublicate is not a typo)
         message = parse_SOAP_response(
             repeat_record.repeater.url,
             repeat_record.repeater.operation,
-            response
+            response,
         )
         try:
-            if isinstance(message, basestring) and message.isdigit():
-                health_facility_id = self._get_person_locations(payload_doc).pcp
-                nikshay_id = '-'.join([health_facility_id, message])
-                update_case(
-                    payload_doc.domain,
-                    payload_doc.case_id,
-                    {
-                        "private_nikshay_registered": "true",
-                        "nikshay_id": nikshay_id,
-                        "private_nikshay_error": "",
-                    },
-                    external_id=nikshay_id,
-                )
-            else:
-                self.handle_failure(message, payload_doc, repeat_record)
-        except NikshayResponseException as e:
-            _save_error_message(payload_doc.domain, payload_doc.case_id, unicode(e.message))
+            health_facility_id = self._get_person_locations(payload_doc).pcp
+        except NikshayLocationNotFound as e:
+            self.handle_exception(e, repeat_record)
+        else:
+            nikshay_id = '-'.join([health_facility_id, message])
+            update_case(
+                payload_doc.domain,
+                payload_doc.case_id,
+                {
+                    "private_nikshay_registered": "true",
+                    "nikshay_id": nikshay_id,
+                    "private_nikshay_error": "",
+                },
+                external_id=nikshay_id,
+            )
 
     def handle_failure(self, response, payload_doc, repeat_record):
-        _save_error_message(payload_doc.domain, payload_doc.case_id, unicode(response),
+        message = parse_SOAP_response(
```

Comment:
we can also modify the `send_request` on the mixin and the `perform_SOAP_operation` to control when we need raw response or just parsed response. Basically, If we skip `raw_response=True` zeep will parse the SOAP response and give us just the final value. The reason i picked it to keep it raw here so that we store the complete response and not just the text response. If there is no need to store the complete xml response we can just receive the parsed response here.
And then this won't be needed everywhere and can be made repeater specific.